src/dual_contouring.rs

```diff
@@ -118,7 +118,9 @@ pub mod edge {
     Material: material::T,
     Voxels: voxel_storage::T<Material>,
   {
-    let [b0, b1] = corner_bounds(edge);
+    let bounds = corner_bounds(edge);
+    let b0 = bounds[0];
+    let b1 = bounds[1];
```

Comment:
This looks like a change in behavior. Can you add an assertion that the length of the slice is exactly 2?

Reply:
The signature of `corner_bounds` says it always has the length 2: https://github.com/bfops/rust-isosurface-extraction/pull/2/files/7c64dcae1e5866254161c31aaf901cb6305262ae#diff-90a554edfb593c44610b44e0f7de3876R76

If it wouldnt the original code wouldnt compile anyway as it wouldnt be an exhaustive match

Reply:
if the signature of `corner_bounds` changed to return an array of different length then the original code would fail to compile, but this version would silently change in behavior. That’s why the added compile-time assertion is nice. Changing it to a runtime assertion instead of a compile time check is still a bit sad but okay. A `debug_assert` would be fine.

Reply:
I see

Reply:
Added a type ascription to `bounds`

Reply:
That works too